tests/e2e/e2e_upgrade_gov_and_authz_test.go

```diff
@@ -0,0 +1,276 @@
+	suite.Require().NoError(err)
+	suite.Require().Equal(authzBeforeGrants.Pagination.Total, uint64(len(authzBeforeGrants.Grants)), "expected all grants to have been requested")
+
+	// no kavadist -> gov grants
+	suite.Equal(0, len(authzBeforeGrants.Grants))
+
+	// fetch authz grants after upgrade
+	authzAfterGrants, err := suite.Kava.Authz.Grants(afterUpgradeCtx, &authz.QueryGrantsRequest{Granter: kavadistModuleAcc, Grantee: govModuleAcc, Pagination: &query.PageRequest{Limit: 1000, CountTotal: true}})
+	suite.Require().NoError(err)
+	suite.Require().Equal(authzAfterGrants.Pagination.Total, uint64(len(authzAfterGrants.Grants)), "expected all grants to have been requested")
+
+	// one kavadist -> gov grants
+	suite.Require().Equal(1, len(authzAfterGrants.Grants))
+
+	grant := authzAfterGrants.Grants[0]
+
+	var authorization authz.Authorization
+	suite.Kava.EncodingConfig.Marshaler.UnpackAny(grant.Authorization, &authorization)
+
+	genericAuthorization, ok := authorization.(*authz.GenericAuthorization)
+	suite.Require().True(ok, "expected generic authorization")
+
+	// kavadist allows gov to MsgSend it's funds
+	suite.Equal(sdk.MsgTypeURL(&banktypes.MsgSend{}), genericAuthorization.Msg)
+	// no expiration
+	var expectedExpiration *time.Time
+	suite.Equal(expectedExpiration, grant.Expiration)
+}
+
+func (suite *IntegrationTestSuite) TestModuleAccountGovTransfers() {
+	suite.SkipIfUpgradeDisabled()
+	suite.SkipIfKvtoolDisabled()
+
+	// the module account (authority) that executes the transfers
+	govAcc := sdk.MustAccAddressFromBech32(govModuleAcc)
+
+	// module accounts for gov transfer test cases
+	communityAcc := sdk.MustAccAddressFromBech32(communityModuleAcc)
+	kavadistAcc := sdk.MustAccAddressFromBech32(kavadistModuleAcc)
+
+	testCases := []struct {
+		name     string
+		sender   sdk.AccAddress
+		receiver sdk.AccAddress
+		amount   sdk.Coin
+	}{
+		{
+			name:     "transfer from community to kavadist for incentive rewards",
+			sender:   communityAcc,
+			receiver: kavadistAcc,
+			amount:   ukava(100e6),
+		},
+		{
+			name:     "transfer from kavadist to community",
+			sender:   kavadistAcc,
+			receiver: communityAcc,
+			amount:   ukava(50e6),
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			// create msg exec for transfer between modules
+			msg := banktypes.NewMsgSend(
+				tc.sender,
+				tc.receiver,
+				sdk.NewCoins(tc.amount),
+			)
+			execMsg := authz.NewMsgExec(govAcc, []sdk.Msg{msg})
+
+			// ensure proposal passes
+			passBlock := suite.submitAndPassProposal([]sdk.Msg{&execMsg})
+
+			// get starting balance of the module account to be funded, 1 block
+			// before the proposal passes
+			startingBalance := suite.Kava.QuerySdkForBalancesAtHeight(passBlock-1, tc.receiver)
+			// get ending balance of the receiving account
+			endingBalance := suite.Kava.QuerySdkForBalancesAtHeight(passBlock, tc.receiver)
+
+			paidRewards := suite.getStakingRewardsAtBlock(passBlock)
```

Comment:
When the receiver is kavadist, should paid rewards be included?

I'm thinking if we can't control the exact amount into or out of these accounts, we should parse the logs to check for a transfer event of the correct amount. Then maybe also check the difference in balance is accounted for by all the transfer events.

ok to merge this if blocking mirrornet, and address in a followup pr